src/Middleware/RequestLimiter/src/IPAggregatedRateLimiter.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Collections.Concurrent;
+using System.Net;
+using System.Threading;
+using System.Threading.ResourceLimits;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.RequestLimiter
+{
+    // TODO: update implementation with WaitAsync and use MemoryCache instead of ConcurrentDictionary
```

Comment:
Why?

Reply:
It would be closer to a "remote store" based limiter example. Something like a RedisRateLimiter we envision. It's for illustration purposes.

Reply:
I don't think we should think about that. We need to do what's best. I worry that we'll lose access to APIs that IMemoryCache hide.

Reply:
Fair enough. Though I'm curious what you think we lose by using IMemoryCache APIs. I think it depends on how we use it.